### PR TITLE
Adding e2e compatibility tests for ZK ACL branch

### PR DESCRIPTION
**Pull request by rahulrane50**
### Description
Adding e2e compatible tests for zookeeper ZK ACL dev branch

### Tests
 Tried this with manual build and it works fine.

The following is the result of the "mvn test" command on the appropriate module:
(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)
My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:
(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)
In case of new functionality, my PR adds documentation in the following wiki page:
(Link the GitHub wiki you added)

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      - name: Download ZooKeeper ${{ matrix.zk }}
+        if: matrix.zk != 'nightly'
+        run: |
+          curl -O https://archive.apache.org/dist/zookeeper/zookeeper-${{ matrix.zk }}/apache-zookeeper-${{ matrix.zk }}-bin.tar.gz
+          tar -xzvf apache-zookeeper-${{ matrix.zk }}-bin.tar.gz
+      - name: Test ZooKeeper nightly server and ${{ matrix.zk }} client
+        if: matrix.zk != 'nightly'
+        run: tools/ci/test-connectivity.py --server . --client apache-zookeeper-${{ matrix.zk }}-bin
+        env:
+          ZOOCFG: zoo_sample.cfg
```
Comment (mgao0) [inline .github/workflows/e2e.yaml:68]: If we are testing for zk acl branch, should we use the zk acl associated config values?

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      - name: Test ZooKeeper ${{ matrix.zk }} server and nightly client
+        if: matrix.zk != 'nightly'
+        run: tools/ci/test-connectivity.py --server apache-zookeeper-${{ matrix.zk }}-bin --client .
+        env:
+          ZOOCFG: zoo_sample.cfg
```
Comment (mgao0) [inline .github/workflows/e2e.yaml:73]: Same question as above

File: .github/workflows/e2e.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      - name: Test ZooKeeper nightly server and client
+        if: matrix.zk == 'nightly'
+        run: tools/ci/test-connectivity.py --server . --client .
+        env:
+          ZOOCFG: zoo_sample.cfg
```
Comment (mgao0) [inline .github/workflows/e2e.yaml:78]: Same question as above

File: tools/ci/test-connectivity.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
```
Comment (mgao0) [inline tools/ci/test-connectivity.py:1]: For file path, we already have [zookeeper-compatibility-tests](https://github.com/linkedin/zookeeper/tree/9b00a86a887aa321ddfe9783461cec8d711eb3b8/zookeeper-compatibility-tests) package, shall we just put this script under that package? If necessary, we can rename the package name for a wider coverage

**Comment by mgao0** [comment]
Thanks for explaining that this is actually a commit from upstream. https://github.com/apache/zookeeper/commit/609afd1281ace4d29aabfb48628aa9eead56c4bf
I have a thought: instead of addressing my comments and merging the PR, let's just cherry pick the commit from upstream, to avoid any confusions or code conflicts with upstream in the future. @rahulrane50 What do you think?

**Comment by rahulrane50** [comment]
Sounds good. As per offline discussion, we should include zookeeper compatibility tests and port e2e tests from zookeeper. I will be closing this PR for now.

**Pull request closed without merging**